apps/builder/app/builder/features/topbar/publish.tsx

```diff
@@ -270,6 +271,8 @@ const Publish = ({
     }
   }, [isPublishing, refresh]);
 
+  const isPublishInProgress = publishState !== "idle" || isPublishing;
```

Comment:
One `isPublishing` is calculated based on data, the other is that publsih action started. With optimistic update systems this will not be an option as data can be changed optimistically and synchronously with operation has started. For now I don't know how to do above better without combining different data sources for state

Reply:
then we need better names to distinguish the two clearly